subworkflows/local/prokka_subsets.nf

```diff
@@ -19,11 +19,11 @@ workflow PROKKA_SUBSETS {
             .set { ch_prokka }
         PROKKA ( ch_prokka, [], [] )
         ch_versions = ch_versions.mix(PROKKA.out.versions)
-        GFF_CAT ( PROKKA.out.gff )
+        GFF_CAT ( PROKKA.out.gff.collect().map { [ [ id: 'prokka' ], it[1] ] } )
         ch_versions = ch_versions.mix(GFF_CAT.out.versions)
-        FAA_CAT ( PROKKA.out.faa )
+        FAA_CAT ( PROKKA.out.faa.collect().map { [ [ id: 'prokka' ], it[1] ] } )
         ch_versions = ch_versions.mix(FAA_CAT.out.versions)
-        FNA_CAT ( PROKKA.out.fna )
+        FNA_CAT ( PROKKA.out.fna.collect().map { [ [ id: 'prokka' ], it[1] ] } )
```

Comment:
Now, this looks odd. From your code it looks like the three channels already had a map (probably containing the assembly name) (since you're using `it[1]`). My guess is that this PR addresses the duplication of the hmm process (issue #83) but not issue #85. I think you should just `collect()` here and not `map()` and fix #83 .

To address #85 you will have to make sure that what these three calls _emit_ contain the map. (If we're fancy, adding `.prokka` to the `id` that's already there would be nice. We talked about this yesterday.)